genestack_client/files_util.py

```diff
@@ -551,3 +553,201 @@ def add_checksums(self, app_file, expected_checksums):
+        Searches only in ``parent`` folder if provided, otherwise everywhere.
+
+        Returns a list of dictionaries with complete information about each of the specified files.
+        Dictionaries have the same structure as dictionaries
+        returned by :py:meth:`~genestack_client.FilesUtil.get_infos` method.
+
+        :param parent: accession of parent folder
+            (if not provided, files will be searched in all folders)
+        :type parent: str
+        :param name: file name (optional)
+        :type name: str
+        :param metainfo: metainfo attributes to build search query
+        :type metainfo: Metainfo
+        :param offset: offset to start retrieving samples. Must be >= 0
+        :type offset: int
+        :param limit: maximum number of files to retrieve. Must be <= 100.
+        :type limit: int
+        :return: list of file info dictionaries
+        :rtype: list
+        """
+        return self.__find_files('sample', parent, name, metainfo, offset, limit)
+
+    def find_studies(self, parent=None, name=None, metainfo=None, offset=0, limit=MAX_SEARCH_LIMIT):
+        """
+        Search studies by name and metainfo.
+        Searches only in ``parent`` folder if provided, otherwise everywhere.
+
+        Returns a list of dictionaries with complete information about each of the specified files.
+        Dictionaries have the same structure as dictionaries
+        returned by :py:meth:`~genestack_client.FilesUtil.get_infos` method.
+
+        :param parent: accession of parent folder
+            (if not provided, files will be searched in all folders)
+        :type parent: str
+        :param name: file name (optional)
+        :type name: str
+        :param metainfo: metainfo attributes to build search query
+        :type metainfo: Metainfo
+        :param offset: offset to start retrieving samples. must be >= 0
+        :type offset: int
+        :param limit: maximum number of files to retrieve. Must be <= 100.
+        :type limit: int
+        :return: list of file info dictionaries
+        :rtype: list
+        """
+        return self.__find_files('study', parent, name, metainfo, offset, limit)
+
+    def get_metainfo(self, accession):
+        """
+        Get file metainfo dictionary.
+        Returns dictionary that maps metainfo keys to lists of dictionaries with the following fields:
+
+            - Metainfo value type
+            - Metainfo value
+
+        Example dictionary::
+
+            {
+                'key1': [
+                    {
+                        'type': 'string',
+                        'value': 'value1'
+                    },
+                    {
+                        'type': 'string',
+                        'value': 'value2'
+                    }
+                ]
+            }
+
+        :param accession: file accession
+        :type accession: str
+        :return: file metainfo dictionary
+        :rtype: dict
```

Comment:
You can specify complex return type like `dict[str, list[dict[str, object]]]`

See https://www.jetbrains.com/help/pycharm/2016.1/type-hinting-in-pycharm.html#legacy